cdc/sink/producer/kafka/kafka.go

```diff
@@ -395,80 +409,121 @@ func (k *kafkaSaramaProducer) run(ctx context.Context) error {
+		if ok {
+			if a, ok := info.ConfigEntries["max.message.bytes"]; ok {
+				topicMaxMessageBytes, err := strconv.Atoi(*a)
+				if err != nil {
+					return err
+				}
+				if topicMaxMessageBytes < config.MaxMessageBytes {
+					return cerror.ErrKafkaInvalidConfig.GenWithStack(
+						"topic already exist, and max.message.size(%d) less than max-message-size(%d)",
+						topicMaxMessageBytes, config.MaxMessageBytes)
+				}
+			}
+
+			config.PartitionNum = realPartitionCount
+			log.Warn("topic already exist", zap.String("topic", config.TopicName))
+			return nil
 		}
-	} else {
-		if partitionNum == 0 {
-			partitionNum = defaultPartitionNum
-			log.Warn("topic not found and partition number is not specified, using default partition number", zap.String("topic", topic), zap.Int32("partition_num", partitionNum))
+		// this could happen if user does not specify the `partition-num` in the sink uri.
+		if config.PartitionNum == 0 {
+			config.PartitionNum = defaultPartitionNum
+			log.Warn("partition-num is not set, use the default partition count",
+				zap.String("topic", config.TopicName), zap.Int32("partitions", config.PartitionNum))
 		}
-		log.Info("create a topic", zap.String("topic", topic),
-			zap.Int32("partition_num", partitionNum),
-			zap.Int16("replication_factor", config.ReplicationFactor))
-		err := admin.CreateTopic(topic, &sarama.TopicDetail{
-			NumPartitions:     partitionNum,
+		maxMessageBytes := strconv.Itoa(config.MaxMessageBytes)
+		err := admin.CreateTopic(config.TopicName, &sarama.TopicDetail{
+			NumPartitions:     config.PartitionNum,
 			ReplicationFactor: config.ReplicationFactor,
+			ConfigEntries: map[string]*string{
+				"max.message.bytes": &maxMessageBytes,
+			},
 		}, false)
-		// TODO idenfity the cause of "Topic with this name already exists"
+		// TODO identify the cause of "Topic with this name already exists"
 		if err != nil && !strings.Contains(err.Error(), "already exists") {
-			return 0, cerror.WrapError(cerror.ErrKafkaNewSaramaProducer, err)
+			return err
 		}
+		return nil
+	}
+
+	// if `auto-create-topic` is disabled by user, we would assume topic should have already created
+	if !ok {
+		return cerror.ErrKafkaInvalidConfig.GenWithStack("auto-create-topic is false, and topic not found")
+	}
+
+	if config.PartitionNum == 0 {
+		config.PartitionNum = realPartitionCount
+		return nil
+	}
+
+	if config.PartitionNum < realPartitionCount {
+		log.Warn("partition count assigned in sink-uri is less than that of topic, some partitions would not have messages to be dispatched",
+			zap.Int32("sink-uri partitions", config.PartitionNum), zap.Int32("topic partitions", realPartitionCount))
+		return nil
+	}
+
+	// for topic already exist, we just make sure that the user specified partition count is not greater than the real
+	// partition count, since we would dispatch messages to different partitions, this could prevent potential message loss.
```

Comment:
Please reword this sentence. `since we would dispatch messages to different partitions` does not have a clear meaning.